discovery-infra/tests/conftest.py

```diff
@@ -69,19 +83,23 @@ def get_api_client(offline_token=env_variables['offline_token'], **kwargs):
     return assisted_service_api.create_client(url, offline_token, **kwargs)
 
 
+@pytest.fixture(scope="session")
+def is_dev_env():
+    return not is_qe_env()
+
+
 @pytest.fixture(scope="session")
 def setup_node_controller():
     logging.info(f'--- SETUP --- node controller\n')
-    controller = nodeController(**env_variables)
-    controller.prepare_nodes()
-    yield controller
+    if is_qe_env():
+        controller = nodeController(**env_variables)
+        controller.prepare_nodes()
+        yield controller
+    yield nodeController
     logging.info(f'--- TEARDOWN --- node controller\n')
-    controller.destroy_all_nodes()
+    if is_qe_env():
+        controller.destroy_all_nodes()
 
-@pytest.fixture(scope="session", autouse=True)
```

Comment:
Why did you remove this?

Reply:
I see running with pytest-xdist creates a new session for each test:
https://github.com/pytest-dev/pytest-xdist/issues/271

Reply:
Not working as multi run uses different process for each test run. Than it run at the first of each test which caused a raise issues..need to make this setup in the beginning of each RUN. Therefore added to Makefile